### PR TITLE
Simplify and improve the local developer experience for pushing images

### PR DESCRIPTION
**Pull request by owainlewis**
This change makes some simplifications to the Makefile around pushing development images.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,19 +18,16 @@ ifdef WERCKER
...
     TENANT := "spinnaker"
 endif
 
 ROOT_DIR        := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
 PKG             := github.com/oracle/mysql-operator
-REGISTRY        := iad.ocir.io/$(TENANT)
+REGISTRY        := iad.ocir.io
 SRC_DIRS        := cmd pkg test/examples
-REGISTRY_STRING := $(subst /,_,$(REGISTRY))
 CMD_DIRECTORIES := $(sort $(dir $(wildcard ./cmd/*/)))
```
Comment (prydie) [inline Makefile]: No longer used

File: Makefile [hunk 2]
```diff
@@ -18,19 +18,16 @@ ifdef WERCKER
...
 COMMANDS        := $(CMD_DIRECTORIES:./cmd/%/=%)
```
Comment (prydie) [inline Makefile]: No longer used

File: Makefile [hunk 3]
```diff
@@ -18,19 +18,16 @@ ifdef WERCKER
...
-CONTAINER_FILES := $(addprefix .container-$(REGISTRY_STRING)-,$(addsuffix -$(VERSION),$(COMMANDS)))
-PUSH_FILES      := $(addprefix .push-$(REGISTRY_STRING)-,$(addsuffix -$(VERSION),$(COMMANDS)))
 
 ARCH    := amd64
```
Comment (prydie) [inline Makefile]: Should be `?=`

File: Makefile [hunk 4]
```diff
@@ -18,19 +18,16 @@ ifdef WERCKER
...
 OS      := linux
```
Comment (prydie) [inline Makefile]: Should be `?=`

File: Makefile [hunk 5]
```diff
@@ -18,19 +18,16 @@ ifdef WERCKER
...
     NEW_NAMESPACE ?= e2e-${USER}
-    VERSION := ${USER}-$(shell date +%Y%m%d%H%M%S)
+    VERSION := ${USER}-$(shell git describe --always --dirty)
     TENANT := "spinnaker"
 endif
 
 ROOT_DIR        := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
 PKG             := github.com/oracle/mysql-operator
-REGISTRY        := iad.ocir.io/$(TENANT)
+REGISTRY        := iad.ocir.io
```
Comment (prydie) [inline Makefile]: Should be `?=`

File: Makefile [hunk 6]
```diff
@@ -18,19 +18,16 @@ ifdef WERCKER
     TENANT := "oracle"
 else
     NEW_NAMESPACE ?= e2e-${USER}
-    VERSION := ${USER}-$(shell date +%Y%m%d%H%M%S)
+    VERSION := ${USER}-$(shell git describe --always --dirty)
     TENANT := "spinnaker"
 endif
 
 ROOT_DIR        := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
```
Comment (prydie) [inline Makefile]: No longer used

File: Makefile [hunk 7]
```diff
@@ -18,19 +18,16 @@ ifdef WERCKER
     TENANT := "oracle"
 else
     NEW_NAMESPACE ?= e2e-${USER}
-    VERSION := ${USER}-$(shell date +%Y%m%d%H%M%S)
+    VERSION := ${USER}-$(shell git describe --always --dirty)
```
Comment (prydie) [inline Makefile]: Should be `?=`

File: Makefile [hunk 8]
```diff
@@ -18,19 +18,16 @@ ifdef WERCKER
     TENANT := "oracle"
 else
     NEW_NAMESPACE ?= e2e-${USER}
```
Comment (prydie) [inline Makefile]: No longer used

File: Makefile [hunk 9]
```diff
@@ -18,19 +18,16 @@ ifdef WERCKER
...
-    VERSION := ${USER}-$(shell date +%Y%m%d%H%M%S)
+    VERSION := ${USER}-$(shell git describe --always --dirty)
     TENANT := "spinnaker"
```
Comment (prydie) [inline Makefile]: Should be `?=`

File: Makefile [hunk 10]
```diff
@@ -102,14 +91,7 @@ lint:
...
-clean: container-clean bin-clean
-
-.PHONY: container-clean
-container-clean:
-	rm -rf .container-* .push-* dist
-
-.PHONY: bin-clean
-bin-clean:
+clean:
 	rm -rf .go bin
```
Comment (prydie) [inline Makefile]: No need to rm `.go` as it's not created any more AFAIK

File: Makefile [hunk 11]
```diff
@@ -75,23 +72,15 @@ build: dist build-dirs Makefile
...
 # Note: Only used for development, i.e. in CI the images are pushed using Wercker.
 .PHONY: push
-push: $(PUSH_FILES)
-.push-$(REGISTRY_STRING)-%-$(VERSION): .container-$(REGISTRY_STRING)-%-$(VERSION)
-	@echo Pushing container: $*
-	@docker login -u '$(DOCKER_REGISTRY_USERNAME)' -p '$(DOCKER_REGISTRY_PASSWORD)' $(REGISTRY)
-	@docker push $(REGISTRY)/$*:$(VERSION)
-	@docker images -q $(REGISTRY)/$*:$(VERSION) > $@
+push: build
+	@docker build --build-arg=http_proxy --build-arg=https_proxy -t $(REGISTRY)/$(TENANT)/mysql-operator:$(VERSION) -f docker/mysql-operator/Dockerfile .
```
Comment (prydie) [inline Makefile]: Would break these up over multiple lines

File: Makefile (hunk 10 above)
Comment (owainlewis) [reply Makefile]: Still created on my machine. Will remove after fixing the build process in a subsequent PR.

File: Makefile (hunk 2 above)
Comment (owainlewis) [reply Makefile]: Still used / referenced. Will remove in a subsequent PR when fixing the build step to keep things clean.

File: Makefile (hunk 11 above)
Comment (owainlewis) [reply Makefile]: Agree. Split into a separate step for hygiene and broken into multiple lines.

**Pull request merged**